src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -501,10 +523,27 @@ public void ProduceContinue()
             }
         }
 
-        public async Task ProduceStartAndFireOnStarting(bool immediate = true)
+        public Task ProduceStartAndFireOnStarting(bool immediate = true)
         {
-            if (_responseStarted) return;
+            if (_responseStarted) return TaskUtilities.CompletedTask;
+
+            if (_onStarting != null)
+            {
+                return FireOnStartingProduceStart(immediate: immediate);
+            }
 
+            if (_applicationException != null)
```

Comment:
No; its just surfaced here for the non-async path; previous gets triggered if it goes down the async path of `FireOnStartingProduceStart` 3 lines above.